team/tests/org.eclipse.compare.tests/src/org/eclipse/compare/tests/DocLineComparatorTest.java

```diff
@@ -258,15 +259,67 @@ public boolean canCacheFilteredRegions() {
 		for (int i = 0; i < docs.length; i++)
 			for (int j = i + 1; j < docs.length; j++)
 				for (ICompareFilter[] filter : filters) {
-					l = new DocLineComparator(docs[i], null, false, filter, 'L');
-					r = new DocLineComparator(docs[j], null, false, filter, 'R');
+					l = new DocLineComparator(docs[i], null, false, filter, 'L', Optional.empty());
+					r = new DocLineComparator(docs[j], null, false, filter, 'R', Optional.empty());
 					Assert.assertFalse(l.rangesEqual(0, r, 0));
-					l = new DocLineComparator(docs[i], null, true, filter, 'L');
-					r = new DocLineComparator(docs[j], null, true, filter, 'R');
+					l = new DocLineComparator(docs[i], null, true, filter, 'L', Optional.empty());
+					r = new DocLineComparator(docs[j], null, true, filter, 'R', Optional.empty());
 					Assert.assertTrue(l.rangesEqual(0, r, 0));
 				}
 	}
 
+	@Test
+	public void whiteSpaceInStringLiteralWithEmptyIgnoreWhitespaceContributor() {
+		DocLineComparator left = new DocLineComparator(new Document("str = \"Hello World\""), null, true, null, 'L', Optional.empty());
+		DocLineComparator right = new DocLineComparator(new Document("str = \"HelloWorld\""), null, true, null, 'R', Optional.empty());
+		Assert.assertTrue(left.rangesEqual(0, right, 0));
+		Assert.assertTrue(right.rangesEqual(0, left, 0));
+	}
+
+	@Test
+	public void whiteSpaceInStringLiteralWithSimpleIgnoreWhitespaceContributor() {
+		rangesEqualWithSimpleIgnoreWhitespaceContrbutor(//
+				"str = \"HelloWorld\"",//
+				"str = \"HelloWorld\"");
+
+		rangesEqualWithSimpleIgnoreWhitespaceContrbutor(//
+				"str =     \"HelloWorld\"   ",//
+				"str = \"HelloWorld\"");
+
+		rangesNotEqualWithSimpleIgnoreWhitespaceContrbutor(//
+				"str = \"Hello World\"",//
+				"str = \"HelloWorld\"");
+
+		rangesNotEqualWithSimpleIgnoreWhitespaceContrbutor(//
+				"str = \" HelloWorld \"",//
+				"str = \"HelloWorld\"");
+
+		rangesNotEqualWithSimpleIgnoreWhitespaceContrbutor(//
+				"str =         \" HelloWorld \"     ",//
+				"str = \"HelloWorld \"");
+	}
+
+
+	private void rangesNotEqualWithSimpleIgnoreWhitespaceContrbutor(String leftSource,String rightSource) {
+		Document leftDocument= new Document(leftSource);
+		DocLineComparator left = new DocLineComparator(leftDocument, null, true, null, 'L', Optional.of(new SimpleIgnoreWhitespaceContributor(leftDocument)));
+
+		Document rightDocument = new Document(rightSource);
+		DocLineComparator right = new DocLineComparator(rightDocument, null, true, null, 'R', Optional.of(new SimpleIgnoreWhitespaceContributor(rightDocument)));
+		Assert.assertFalse(left.rangesEqual(0, right, 0));
```

Comment:
It would be better if you could add an assertion message (to all of the asserts). With the current state we only get "expected false but was true" as the output on failure. Something more speaking would be nice.

Reply:
done with 9376205